pints/tests/test_slice_stepout.py

```diff
@@ -0,0 +1,350 @@
+# -*- coding: utf-8 -*-
+#
+# Tests the basic methods of the Slice Sampling routine.
+#
+# This file is part of PINTS.
+#  Copyright (c) 2017-2019, University of Oxford.
+#  For licensing information, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+
+import unittest
+import numpy as np
+
+import pints
+import pints.toy
+
+debug = False
+
+
+class TestSliceStepout(unittest.TestCase):
+    """
+    Tests the basic methods of the Slice Sampling with Stepout routine.
+
+    Please refer to the _slice_stepout.py script in ..\_mcmc\_slice_stepout.py
+    """
+    def test_initialisation(self):
+        """
+        Tests whether all instance attributes are initialised correctly.
+        """
+        # Create mcmc
+        x0 = np.array([2, 4])
+        mcmc = pints.SliceStepoutMCMC(x0)
+
+        # Test attributes initialisation
+        self.assertFalse(mcmc._running)
+        self.assertFalse(mcmc._ready_for_tell)
+        self.assertFalse(mcmc._first_expansion)
+        self.assertFalse(mcmc._interval_found)
+        self.assertFalse(mcmc._set_l)
+        self.assertFalse(mcmc._set_r)
+        self.assertFalse(mcmc._init_left)
+        self.assertFalse(mcmc._init_right)
+
+        self.assertEqual(mcmc._current, None)
+        self.assertEqual(mcmc._current_log_pdf, None)
+        self.assertEqual(mcmc._current_log_y, None)
+        self.assertEqual(mcmc._proposed, None)
+        self.assertEqual(mcmc._l, None)
+        self.assertEqual(mcmc._r, None)
+        self.assertEqual(mcmc._temp_l, None)
+        self.assertEqual(mcmc._temp_r, None)
+        self.assertEqual(mcmc._k, None)
+        self.assertEqual(mcmc._j, None)
+        self.assertEqual(mcmc._fx_l, None)
+        self.assertEqual(mcmc._fx_r, None)
+
+        self.assertEqual(mcmc._m, 50)
+        self.assertEqual(mcmc._active_param_index, 0)
```

Comment:
These are OK for now, but ideally a unit test should test the API of a class works the way it should: for example:

```
x.set_value(5)
self.assertEqual(x.get_value(), 5)
```

How the object stores that value internally is not something you test. In fact, it should be easy to change the way things are handled internally without changing the test status: Just as long as it still works!

So internally I could have `self._value = 5`, or `self._super_value = 5` or `self._values = [5]`. It doesn't matter for the test, as long as it acts correctly _via the API_